### PR TITLE
poc for object mapper

### PR DESCRIPTION
**Pull request by rdrozdov-tc**

### CODE REVIEW

File: core/src/main/java/io/github/tcdl/msb/adapters/mock/MockAdapter.java [hunk 1]
```diff
@@ -50,9 +52,12 @@ public MockAdapter(String topic, Queue<ExecutorService> activeConsumerExecutors)
     public void publish(String jsonMessage) {
         LOG.debug("Received request {}", jsonMessage);
         try {
-            Message incomingMessage = Utils.fromJson(jsonMessage, Message.class);
-            pushRequestMessage(incomingMessage);
-        } catch (JsonConversionException e) {
+            JsonNode messageAsNode = jsonReader.read(jsonMessage);
+            if (messageAsNode.has("topics")) {
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/msb/adapters/mock/MockAdapter.java]: what if wit doesn't contain topics? Should we throw an exception?

Comment (rdrozdov-tc) [inline core/src/main/java/io/github/tcdl/msb/adapters/mock/MockAdapter.java]: Yes, you are right

File: core/src/main/java/io/github/tcdl/msb/api/MsbContextBuilder.java [hunk 2]
```diff
@@ -108,4 +135,30 @@ public void run() {
         
         return msbContext;
     }
+
+    public ObjectMapper buildMessageMapper(ObjectMapper payloadMapper) {
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/msb/api/MsbContextBuilder.java:139]: Can we cover this method with unit test?

File: core/src/main/java/io/github/tcdl/msb/api/MsbContextBuilder.java [hunk 3]
```diff
@@ -63,7 +78,17 @@ public MsbContextBuilder withDefaultChannelMonitorAgent(boolean withDefaultChann
         this.withDefaultChannelMonitorAgent = withDefaultChannelMonitorAgent;
         return this;
     }
-    
+
+    /**
+     * Specifies payload mapper to serialize/deserialize message payload
+     * @param payloadMapper if not provided default mapper will be used
```
Comment (rkatsyuryna) [inline core/src/main/java/io/github/tcdl/msb/api/MsbContextBuilder.java]: mapper seems to be not so obvious name(try to google for "java mapper"). Json ObjectMapper or  ObjectMapper will be better

File: acceptance/src/main/java/io/github/tcdl/msb/acceptance/MsbTestHelper.java [hunk 4]
```diff
@@ -49,6 +51,10 @@ public MsbContext getContext() {
         return context;
     }
 
+    public ObjectMapper getObjectMapper() {
+        return ((MsbContextImpl)(getContext())).getMessageMapper();
```
Comment (vso-tc) [inline acceptance/src/main/java/io/github/tcdl/msb/acceptance/MsbTestHelper.java:55]: why so many braces?

**Pull request merged**